doc/jemalloc.xml.in

```diff
@@ -919,13 +919,15 @@ mallctl("arena." STRINGIFY(MALLCTL_ARENAS_ALL) ".decay",
       <varlistentry id="opt.metadata_thp">
         <term>
           <mallctl>opt.metadata_thp</mallctl>
-          (<type>bool</type>)
+          (<type>const char *</type>)
           <literal>r-</literal>
         </term>
-        <listitem><para>If true, allow jemalloc to use transparent huge page
-        (THP) for internal metadata (see <link
-        linkend="stats.metadata">stats.metadata</link> for details).  This
-        option is disabled by default.</para></listitem>
+        <listitem><para>Controls whether to allow jemalloc to use transparent
+        huge page (THP) for internal metadata (see <link
+        linkend="stats.metadata">stats.metadata</link>).  <quote>always</quote>
+        allows such usage.  <quote>lazy</quote> uses no THP initially, but
+        switches to using THP once metadata usage reaches THP size.  The default
+        is <quote>disabled</quote>.</para></listitem>
```

Comment:
This is the output:

opt.metadata_thp (const char *) r-
Controls whether to allow jemalloc to use transparent huge page (THP) for internal metadata (see stats.metadata). “always” allows such usage. “lazy” uses no THP initially, but switches to using THP once metadata usage reaches THP size. The default is “disabled”.

I feel we should be vague enough to allow internal changes in the future, but still make it fairly clear. Suggestions @davidtgoldblatt ?

Reply:
Maybe

    "lazy" uses no THP initially, but may begin to do so when internal heuristics indicate it could be advantageous.

?

Reply:
Although, at that point, might as well just name it "auto".

Reply:
I think "lazy" sort of implies a particular strategy of preallocating and then madvising once we hit hugepage size, whereas "auto" is more vague.

Reply:
I'm sold on "auto". How about "disabled, auto, enabled", since "always" could be confusing (we do have metadata not covered by this)?

Reply:
Personally I find "always" less confusing than "enabled". To me, both heuristic THP and forced THP could be accurately described by "enabled"; "always" would make it unambiguous to a reader that the hugepages are a sure thing (especially if reading it out of a malloc conf instead of the man page). I wouldn't take it as an API guarantee that metadata definitely lives in hugepages or anything.

I don't have a super strong opinion though; feel free to disregard unless, say, @djwatson disagrees.

Reply:
I don't have a strong feeling here either. My initial thought was, we have a three way switch with "on", "off" plus "auto" being either on or off depending on our heuristic. Well, naming is hard. Opinions @djwatson ?

Reply:
Well I changed my vote. I'm going with "always" now.